sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/utils/ResponseHandler.java

```diff
@@ -148,24 +147,40 @@ private static void handleRequestReturn(BlockStmt body, TypeMirror returnType, j
             addReturnStatement(body, returnIsResponse, "networkResponse.getValue().toStream()");
         } else if (TypeUtil.isTypeOrSubTypeOf(entityType, BinaryData.class)) {
             // Return type is a BinaryData. Return the network response body.
-            // DO NOT close the network response for this return as it will result in the BinaryData either being
-            // closed or invalid when it is returned.
+            // DO NOT close the network response for this return as it will result in the BinaryData either being closed or invalid when it is returned.
             if (returnIsResponse) {
+                if (returnType instanceof DeclaredType) {
+                    DeclaredType declaredType = (DeclaredType) returnType;
+                    if (!declaredType.getTypeArguments().isEmpty()
+                        && ((TypeElement) ((DeclaredType) declaredType.getTypeArguments().get(0)).asElement())
+                            .getQualifiedName()
+                            .contentEquals(List.class.getCanonicalName())) {
+                        // Response<List<BinaryData>> or other generics
+                        handleDeclaredTypes(body, returnType, serializationFormatSet, true);
```

Comment:
In this case we'd actually want to close the network response as we're going to eagerly convert it from a stream of content to buffered content (as there's no generic way to have `List<BinaryData>` where each is a segment of the stream).